trace_decoder/src/processed_block_trace.rs

```diff
@@ -89,10 +89,32 @@ impl BlockTrace {
             extra_code_hash_mappings: pre_image_data.extra_code_hash_mappings.unwrap_or_default(),
         };
 
+        let last_tx_idx = self.txn_info.len() - 1;
+
```

Comment:
Hmm... Can this underflow if we have an empty block? Maybe `saturating_sub` would fix this?

Reply:
Hmm and actually I think I need to handle withdrawals on empty payloads. Not that it matters in real life much but for test chains this may be a pb. will fix both.

Reply:
In case the last payload is a dummy, it hasn't been processed by the incoming jerigon payload so I added logic to insert withdrawal state accesses. Also fixed the underflow here (although it wouldn't even enter the loop so wouldn't have been a big deal, but I like it better out of consistency).